sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImplBuilder.java

```diff
@@ -48,22 +85,130 @@ public KeyVaultAccessControlClientImplBuilder serializerAdapter(SerializerAdapte
+     */
+    public KeyVaultAccessControlClientImplBuilder httpLogOptions(HttpLogOptions httpLogOptions) {
+        this.httpLogOptions = httpLogOptions;
+        return this;
+    }
+
+    /*
+     * The retry policy that will attempt to retry failed requests, if
+     * applicable.
+     */
+    private RetryPolicy retryPolicy;
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the KeyVaultAccessControlClientImplBuilder.
+     */
+    public KeyVaultAccessControlClientImplBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /*
+     * The list of Http pipeline policies to add.
+     */
+    private final List<HttpPipelinePolicy> pipelinePolicies;
+
+    /**
+     * Adds a custom Http pipeline policy.
+     *
+     * @param customPolicy The custom Http pipeline policy to add.
+     * @return the KeyVaultAccessControlClientImplBuilder.
+     */
+    public KeyVaultAccessControlClientImplBuilder addPolicy(HttpPipelinePolicy customPolicy) {
+        pipelinePolicies.add(customPolicy);
+        return this;
+    }
+
     /**
      * Builds an instance of KeyVaultAccessControlClientImpl with the provided parameters.
      *
      * @return an instance of KeyVaultAccessControlClientImpl.
      */
     public KeyVaultAccessControlClientImpl buildClient() {
+        if (apiVersion == null) {
+            this.apiVersion = "7.2";
+        }
         if (pipeline == null) {
-            this.pipeline =
-                    new HttpPipelineBuilder()
-                            .policies(new UserAgentPolicy(), new RetryPolicy(), new CookiePolicy())
-                            .build();
+            this.pipeline = createHttpPipeline();
         }
         if (serializerAdapter == null) {
             this.serializerAdapter = JacksonAdapter.createDefaultSerializerAdapter();
         }
-        KeyVaultAccessControlClientImpl client = new KeyVaultAccessControlClientImpl(pipeline, serializerAdapter);
+        KeyVaultAccessControlClientImpl client =
+                new KeyVaultAccessControlClientImpl(pipeline, serializerAdapter, apiVersion);
         return client;
     }
+
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+                (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
+        policies.add(
+                new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion, buildConfiguration));
```

Comment:
how does authentication work here ?
Don't see any auth policy

Reply:
We have a `KeyVaultCredentialPolicy` set in our client builders that's identical to that of our other Key Vault libraries. The class you are referencing here is auto-generated, hence why the auth policy is not reflected at this level.